src/features/leda-nft/services/leda-nft.service.ts

```diff
@@ -16,7 +17,12 @@ export default class LedaNftService implements INftService {
     this.contract = await createContract<LedaNFT>(ledaNftAddress.address, ledaNft.abi);
   }
 
+  public async getOwner(tokenId: number): Promise<string | undefined> {
+    return this.contract?.ownerOf(tokenId);
+  }
+
   public async mint(tokenURI: string, royalty: number): Promise<ContractTransaction | undefined> {
+    await this.contract?.setApprovalForAll(marketplaceAddress.address, true);
```

Comment:
This is the leda-nft service, however you are using a marketplace address here as well. Is that expected? Shouldn't we separate things ?

Reply:
This was the one that Gimer and I told you about the issue of approving the nft to be able to list it and do the rest.

Reply:
We need to talk to know where to put it, but that was the solution we found at that time